code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java

```diff
@@ -0,0 +1,113 @@
+/*
+This file is used to provide a template for displaying list of events, that other fragments can use.
+Outstanding Issues:
+ */
+
+package com.example.noram;
+
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import androidx.fragment.app.Fragment;
+
+import com.example.noram.controller.EventArrayAdapter;
+import com.example.noram.controller.EventManager;
+import com.example.noram.model.Attendee;
+import com.example.noram.model.Event;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+
+import java.util.ArrayList;
+
+/**
+ * A template {@link Fragment} for event lists.
+ * Should be used to create fragments that contain lists of events
+ * @maintainer Gabriel
+ * @author Gabriel
+ */
+public abstract class EventListFragmentTemplate extends Fragment {
+    private ArrayList<Event> searchEventDataList; // data list of events' search results
+    EventArrayAdapter searchEventAdapter; // adapter for searchEvent list
+    private final CollectionReference eventRef = MainActivity.db.getEventsRef(); // list of events in database
+
+    protected void setupSearch(ListView searchList, EditText searchInput){
+        // basic connections between lists and adapters
+        searchEventDataList = new ArrayList<>();
+        searchEventAdapter = new EventArrayAdapter(this.getContext(), searchEventDataList);
+        searchList.setAdapter(searchEventAdapter);
+
+        // connect each element of the list to a function
+        searchList.setOnItemClickListener((parent, view, position, id) -> {
+            Event event = searchEventDataList.get(position);
+            searchElementsClick(event);
+        });
+
+        // connect searchbar to listen for user input
+        searchInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

```

Comment:
comment all these plz

Reply:
Don't we use regular comments inside methods (including comments for listeners)?

Reply:
whenever you are defining a function such as beforeTextChanged it needs a javadoc